Sources/Sentry/SentryFileManager.m

```diff
@@ -51,6 +51,17 @@ - (nullable instancetype)initWithOptions:(SentryOptions *)options
                            error:error];
 }
 
+- (void)ensureSentryPath
```

Comment:
Good call, I'll make that change!

Reply:
This wound up leading me to further refactor SentryFileManager to not have an inout error parameter in its init, which removed lots of potential crashes in the test suite and simplified things a bit further. See https://github.com/getsentry/sentry-cocoa/pull/2424/commits/116f72ba87e8b8d7715d76b3266fad99aae4f294 cc @philipphofmann 

This changes the init for SentryClient as well, to no longer be nullable. But I don't see anywhere that checks for a nil client. If that's the case then we should be fine to have this change. There is also the edge case: what if we are able to init the file manager, but then the sentry directory is removed and we are unable to recreate it later? That is what causes a nil to be returned from the file manager's init, so we wouldn't handle the same error consistently later on at runtime. So I think it's a good change to make.

Reply:
If we really do want a failable init for SentryClient (which was achieved by it returning `nil`), we could instead put an inout error paremeter on the init for _SentryClient_, instead of just SentryFileManager, and bubble it up to `SentrySDK` so it's more obvious what's going on there, instead of a bunch of implicit no-ops being called on a nil client object, we could just early return there.